rp2040-hal/src/gpio/pull.rs

```diff
@@ -0,0 +1,57 @@
+use paste::paste;
+
+pub(crate) mod pull_sealed {
+    use super::DynPullType;
+
+    pub trait PullType {
+        fn from(pm: DynPullType) -> Self;
+        fn as_dyn(&self) -> DynPullType;
+    }
+}
+/// Type-level `enum` for pull resitor types.
+pub trait PullType: pull_sealed::PullType {}
+
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+/// Value-level `enum` for pull resistor types.
+pub enum DynPullType {
+    #[allow(missing_docs)]
+    None,
+    #[allow(missing_docs)]
+    Up,
+    #[allow(missing_docs)]
+    Down,
+    /// This enables both pull resistor. This setup is described as bus-keep in the RP2040
+    /// datasheet.
+    Both,
```

Comment:
The bus-keep function is very different from "enable both pull-up and pull-down resistors". Just having two resistors enabled would work as a voltage divider, stabilizing the voltage on the line to some fixed value between 0V and 3.3V, as long as nothing else is driving the line. But bus-keep will pull the line to either 0V or 3.3V, depending on the initial voltage.

Therefore I think the name "Both" as well as the description "enables both pull resistor" are misleading. I'd prefer the naming from the data sheet: Just call it `BusKeep`